airlock/business_logic.py

```diff
@@ -1337,11 +1310,28 @@ def get_or_create_current_request(
 
         # requests for output-checkers, and for archived workspaces and inactive
         # projects are still viewable, check if user has permission to create one
+        # Note: we check this here, as this function is only used when we are
+        # about to modify a request, so do a look ahead check to make sure we can
         permissions.check_user_can_action_request_for_workspace(user, workspace)
```

Comment:
I see there's been a bit of discussion in Slack about this function, and I share the confusion about why we'd check for these permissions when _getting_ a request.  The comment helps, but what happens if we move the check to after the `return request` line?

Reply:
If we do that, the test mentioned in the discussion fails. I discovered this by doing exactly as you suggest.

I'm not sure if this behaviour is load bearing on not. I can investigate further?

Reply:
The failing tests are testing `get_or_create_current_request` directly.  `get_or_create_current_request` is only used by `workspace_add_file_to_request`, and tests for `workspace_add_file_to_request` continue to pass if we move the permissions check.

Additionally, we've got extra permission checks later on in the BLL functions called by `workspace_add_file_to_request`.  As such, the permissions check that protects _getting_ an existing request is not load bearing.

So I'm pretty confident we can move the permissions check, and remove the failing tests.  I think the failing tests are too closely coupled to our current implementation, and don't test anything about the actual behaviour of the system.

Reply:
So, there's another bll functions that are "protected" by this: `add_withdrawn_file_to_request`, but that seems fine also, so I agree.

Reply:
done

Reply:
great